python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -358,3 +360,382 @@ def explain_graph(self, graph, feat, tmp=1.0, training=False, **kwargs):
         self.clear_masks()
 
         return (probs, edge_mask) if training else (probs.data, edge_mask)
+
+
+class HeteroPGExplainer(nn.Module):
```

Comment:
makes sense, it should save lots on repetition since only `explain_graph` and `train_step` require changes

Reply:
before this, do you approve of removing the `feat` parameter completely from [`set_masks`](https://github.com/dmlc/dgl/blob/147f6d243532e52c082623463d69cf444030112c/python/dgl/nn/pytorch/explain/pgexplainer.py#L67-L85) in `PGExplainer` and replacing the `num_nodes` lines with:

```python
num_nodes, _ = feat.shape
```

with

```python
num_nodes = graph.num_nodes()
```

this would make the function usable within the hetero impl. as well

Reply:
Sure.

Reply:
done 👍